test/12_conditional_commit.ts

```diff
@@ -38,379 +63,507 @@ let BosonRouter_Factory: ContractFactory;
-    )) as Contract & BosonRouter;
-
-    contractBSNTokenPrice = (await MockERC20Permit_Factory.deploy(
-      'BosonTokenPrice',
-      'BPRC'
-    )) as Contract & MockERC20Permit;
-
-    contractBSNTokenDeposit = (await MockERC20Permit_Factory.deploy(
-      'BosonTokenDeposit',
-      'BDEP'
-    )) as Contract & MockERC20Permit;
-    contractERC1155NonTransferable =
-      (await ERC1155NonTransferable_Factory.deploy(
-        'https://token-cdn-domain/{id}.json'
-      )) as Contract & ERC1155NonTransferable;
-    contractGate = (await Gate_Factory.deploy(
-      contractAddresses.BosonRouter,
-      contractAddresses.ERC1155NonTransferable
-    )) as Contract & Gate;
-
-    await contractTokenRegistry.deployed();
-    await contractVoucherSets.deployed();
-    await contractVouchers.deployed();
-    await contractVoucherKernel.deployed();
-    await contractCashier.deployed();
-    await contractBosonRouter.deployed();
-    await contractBSNTokenPrice.deployed();
-    await contractBSNTokenDeposit.deployed();
-    await contractERC1155NonTransferable.deployed();
-    await contractGate.deployed();
-
-    await contractVoucherSets.setApprovalForAll(
-      contractVoucherKernel.address,
-      true
-    );
-
-    await contractVouchers.setApprovalForAll(
-      contractVoucherKernel.address,
-      true
-    );
-
-    await contractVoucherKernel.setComplainPeriod(sixtySeconds);
-    await contractVoucherKernel.setCancelFaultPeriod(sixtySeconds);
-
-    await contractTokenRegistry.setTokenLimit(
-      contractBSNTokenPrice.address,
-      constants.TOKEN_LIMIT
-    );
-    await contractTokenRegistry.setTokenLimit(
-      contractBSNTokenDeposit.address,
-      constants.TOKEN_LIMIT
-    );
-
-    await contractTokenRegistry.setETHLimit(constants.ETHER_LIMIT);
-
-    //Map $BOSON token to itself so that the token address can be called by casting to the wrapper interface in the Boson Router
-    await contractTokenRegistry.setTokenWrapperAddress(
-      contractBSNTokenPrice.address,
-      contractBSNTokenPrice.address
-    );
-
-    await contractTokenRegistry.setTokenWrapperAddress(
-      contractBSNTokenDeposit.address,
-      contractBSNTokenDeposit.address
-    );
-
-    await contractBosonRouter.setGateApproval(contractGate.address, true);
-  }
+    )
+  );
+}
+
+async function setupConditionalToken(token: number) {
+  switch (token) {
+    case conditionalTokens.QUEST:
+      // Setup ERC1155NonTransferable as conditional token
+      await contractGate.pause();
+      await contractGate.setConditionalTokenContract(
+        contractERC1155NonTransferable.address,
+        constants.TOKEN_TYPE.MULTI_TOKEN
+      );
+      await contractGate.unpause();
+
+      // Mint an ERC1155NonTransferable token for buyer
+      await contractERC1155NonTransferable.mint(
+        users.buyer.address,
+        constants.NFT_TOKEN_ID,
+        constants.ONE,
+        constants.ZERO_BYTES
+      );
+      //console.log("Setup Quest");
```

Comment:
comment can be removed

Reply:
Had one of these for all the token types. Removed them all